core/src/main/scala/com/salesforce/op/stages/sparkwrappers/specific/OpPredictorWrapper.scala

```diff
@@ -98,12 +100,19 @@ class OpPredictorWrapper[E <: Predictor[Vector, E, M], M <: PredictionModel[Vect
       .set(po, getOutputFeatureName)
       .fit(dataset)
 
-    SparkModelConverter.toOP(model, uid)
+    val wrappedModel = SparkModelConverter.toOP(model, uid)
       .setParent(this)
       .setInput(in1.asFeatureLike[RealNN], in2.asFeatureLike[OPVector])
       .setMetadata(getMetadata())
       .setOutputFeatureName(getOutputFeatureName)
+
+    if (model.isInstanceOf[XGBoostClassificationModel] || model.isInstanceOf[XGBoostRegressionModel]) {
+      wrappedModel.setOutputDF(model.transform(dataset.limit(1)))
```

Comment:
just curious why do we have `.limit(1)` here ?

Reply:
we only need one example for the xgboost mleap save (it has a step that calls .first() to get the vector size)

Reply:
Let's add a comment for it. Looks like this is the only such exception so far.